sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JDBCRDD.scala

```diff
@@ -175,6 +175,27 @@ object JDBCRDD extends Logging {
       requiredColumns: Array[String],
       filters: Array[Filter],
       parts: Array[Partition],
+      options: JDBCOptions): RDD[InternalRow] = {
+    val url = options.url
+    val dialect = JdbcDialects.get(url)
+    val quotedColumns = requiredColumns.map(colName => dialect.quoteIdentifier(colName))
+    new JDBCRDD(
+      sc,
+      dialect.createConnectionFactory(options),
+      pruneSchema(schema, requiredColumns),
+      quotedColumns,
+      filters,
+      parts,
+      url,
+      options)
+  }
+
+  def scanTable(
```

Comment:
why do we have 2 `scanTable`? I think JDBC V2 should either support v1 or v2 filter, not both.

Reply:
```
@Stable
trait PrunedFilteredScan {
  def buildScan(requiredColumns: Array[String], filters: Array[Filter]): RDD[Row]
}
```